extensions/wikia/FacebookTags/FacebookTagParser.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+class FacebookTagParser {
+	/** @var string $tagName */
+	private $tagName;
+
+	public function __construct( string $tagName ) {
+		if ( isset( FacebookTagConstants::TAG_FALLBACKS[$tagName] ) ) {
+			$this->tagName = FacebookTagConstants::TAG_FALLBACKS[$tagName];
+			return;
+		}
+
+		$this->tagName = $tagName;
+	}
+
+	public function __invoke( ...$args ) {
+		/**
+		 * @var string $contents
+		 * @var array $attributes
+		 * @var Parser $parser
+		 * @var PPFrame $frame
+		 */
+		list( $contents, $attributes, $parser, $frame ) = $args;
+
+		$tagAttributes = $this->buildTagAttributes( $attributes );
+
+		return Html::element( 'iframe', $tagAttributes );
+	}
+
+	private function buildTagAttributes( array $attributes ): array {
+		$attributeWhitelist = FacebookTagConstants::SUPPORTED_TAG_ATTRIBUTES;
+		$tagAttributes = Sanitizer::validateAttributes( $attributes, $attributeWhitelist );
+
+		$queryParameters = $this->buildQueryParameters( $attributes );
+		$fbUrl = sprintf( FacebookTagConstants::FB_API_URL, $this->tagName, $queryParameters );
+
+		$tagAttributes += [
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$tagAttributes". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)